src/libPMacc/PMaccConfig.cmake

```diff
@@ -0,0 +1,167 @@
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall")
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-unknown-pragmas")
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wextra")
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-unused-parameter")
+# ICC
+elseif("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Intel")
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall")
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -DBOOST_NO_VARIADIC_TEMPLATES")
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -DBOOST_NO_CXX11_VARIADIC_TEMPLATES")
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -DBOOST_NO_FENV_H")
+# PGI
+elseif("${CMAKE_CXX_COMPILER_ID}" STREQUAL "PGI")
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Minform=inform")
+endif()
+
+################################################################################
+# CUDA
+################################################################################
+find_package(CUDA 5.0 REQUIRED)
+
+if(CUDA_VERSION VERSION_LESS 5.5)
+    message(STATUS "CUDA Toolkit < 5.5 detected. We strongly recommend to still "
+                   "use CUDA 5.5+ drivers (319.82 or higher)!")
+endif(CUDA_VERSION VERSION_LESS 5.5)
+
+set(CUDA_ARCH sm_20 CACHE STRING "Set GPU architecture")
+string(COMPARE EQUAL ${CUDA_ARCH} "sm_10" IS_CUDA_ARCH_UNSUPPORTED)
+string(COMPARE EQUAL ${CUDA_ARCH} "sm_11" IS_CUDA_ARCH_UNSUPPORTED)
+string(COMPARE EQUAL ${CUDA_ARCH} "sm_12" IS_CUDA_ARCH_UNSUPPORTED)
+string(COMPARE EQUAL ${CUDA_ARCH} "sm_13" IS_CUDA_ARCH_UNSUPPORTED)
+
+if(IS_CUDA_ARCH_UNSUPPORTED)
+    message(FATAL_ERROR "Unsupported CUDA architecture ${CUDA_ARCH} specified. "
+                       "SM 2.0 or higher is required.")
+endif(IS_CUDA_ARCH_UNSUPPORTED)
+
+set(CUDA_FTZ "--ftz=false" CACHE STRING "Set flush to zero for GPU")
+
+set(CUDA_MATH --use_fast_math CACHE STRING "Enable fast-math" )
+option(CUDA_SHOW_REGISTER "Show kernel registers and create PTX" OFF)
+option(CUDA_KEEP_FILES "Keep all intermediate files that are generated during internal compilation steps (folder: nvcc_tmp)" OFF)
+option(CUDA_SHOW_CODELINES "Show kernel lines in cuda-gdb and cuda-memcheck" OFF)
+
+if(CUDA_SHOW_CODELINES)
+    set(CUDA_NVCC_FLAGS "${CUDA_NVCC_FLAGS}" --source-in-ptx -Xcompiler -rdynamic -lineinfo)
+    set(CUDA_KEEP_FILES ON CACHE BOOL "activate keep files" FORCE)
+endif(CUDA_SHOW_CODELINES)
+
+set(CUDA_NVCC_FLAGS ${CUDA_NVCC_FLAGS} ${nvcc_flags} -arch=${CUDA_ARCH} ${CUDA_MATH} ${CUDA_FTZ})
+if(CUDA_SHOW_REGISTER)
+    set(CUDA_NVCC_FLAGS "${CUDA_NVCC_FLAGS}" -Xptxas=-v)
+endif(CUDA_SHOW_REGISTER)
+
+if(CUDA_KEEP_FILES)
+    file(MAKE_DIRECTORY "${PROJECT_BINARY_DIR}/nvcc_tmp")
+    set(CUDA_NVCC_FLAGS "${CUDA_NVCC_FLAGS}" --keep --keep-dir "${PROJECT_BINARY_DIR}/nvcc_tmp")
+endif(CUDA_KEEP_FILES)
+
+################################################################################
+# VAMPIR
+################################################################################
+option(VAMPIR_ENABLE "create PMacc vampir support" OFF)
+
+if(VAMPIR_ENABLE)
```

Comment:
sure, for now. see [PIConGPU](https://github.com/ComputationalRadiationPhysics/picongpu/blob/a560077d22cb43c089ac8b0840c62fad11bf0626/src/picongpu/CMakeLists.txt#L106-L209)